### PR TITLE
homework9

### PR DESCRIPTION
**Pull request by Pokosteler**
homework9

### CODE REVIEW

File: src/lesson9/javaBeginnerChapter2/task28/Box.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package lesson9.javaBeginnerChapter2.task28;
```
Comment (TheFallen058) [inline src/lesson9/javaBeginnerChapter2/task28/Box.java]: I don't like this package naming. 
package name should be understandable and it didn't contain upper case.
Ex: javabeginnerchapter2
better name: just jbchapter2

File: src/lesson9/javaBeginnerChapter2/task28/Box.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+public class Box<T> {
+    private T[] array;
```
Comment (TheFallen058) [inline src/lesson9/javaBeginnerChapter2/task28/Box.java]: Your example works, but main idea of the task was to show that we can use T type and not specify some type. As result we can put array of strings in one implementation of Box class and array of intergers in another one.

File: src/lesson9/MyService/AnnotationRunner.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import java.lang.reflect.Method;
+
+public class AnnotationRunner {
+    public static void main(String[] args) throws InvocationTargetException, IllegalAccessException {
+        inspectService(MyService.class);
+    }
+
+    static void inspectService(Class<?> serviceClass) throws IllegalAccessException, InvocationTargetException {
+        if (serviceClass.isAnnotationPresent(Version.class)) {
+            Version ver = (Version) serviceClass.getAnnotation(Version.class);
```
Comment (TheFallen058) [inline src/lesson9/MyService/AnnotationRunner.java]: redundant typecast

File: src/lesson9/MyService/AnnotationRunner.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            System.out.println(ver);
+            MyService service = new MyService();
+            service.setFirstValue(1);
+            service.setSecondValue(2);
+            Method[] methods = serviceClass.getDeclaredMethods();
+            for (Method m : methods) {
+                if (m.getAnnotation(MethodService.class) != null) {
+                    if (m.getAnnotation(MethodService.class).value() == true) {
```
Comment (TheFallen058) [inline src/lesson9/MyService/AnnotationRunner.java]: value() return type is boolean, no need to check for true.

File: src/lesson9/MyService/AnnotationRunner.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                        m.setAccessible(true);
+                        m.invoke(service, null);
```
Comment (TheFallen058) [inline src/lesson9/MyService/AnnotationRunner.java]: args can be left empty, no need to send null

**Review by TheFallen058 — COMMENTED** [review]
Please check my notes

File: src/lesson9/javaBeginnerChapter2/task28/Box.java (hunk 1 above)
Comment (Pokosteler) [reply src/lesson9/javaBeginnerChapter2/task28/Box.java]: fixed

File: src/lesson9/MyService/AnnotationRunner.java (hunk 3 above)
Comment (Pokosteler) [reply src/lesson9/MyService/AnnotationRunner.java]: fixed

File: src/lesson9/MyService/AnnotationRunner.java (hunk 4 above)
Comment (Pokosteler) [reply src/lesson9/MyService/AnnotationRunner.java]: fixed

File: src/lesson9/MyService/AnnotationRunner.java (hunk 5 above)
Comment (Pokosteler) [reply src/lesson9/MyService/AnnotationRunner.java]: fixed

File: src/lesson9/javaBeginnerChapter2/task28/Box.java (hunk 2 above)
Comment (Pokosteler) [reply src/lesson9/javaBeginnerChapter2/task28/Box.java]: fixed

**Review by TheFallen058 — APPROVED** [review]
done

**Pull request merged**